presto-thrift-connector/src/main/java/com/facebook/presto/connector/thrift/util/RetryDriver.java

```diff
@@ -143,29 +154,32 @@ public RetryDriver withClassifier(Function<Exception, Exception> classifier)
                 }
                 catch (InterruptedException ie) {
                     Thread.currentThread().interrupt();
-                    throw propagate(ie);
+                    throw propagate(retryStatus, retryStats, ie);
                 }
             }
         }
     }
 
-    public <V> ListenableFuture<V> runAsync(String callableName, Callable<ListenableFuture<V>> callable)
+    public <V> ListenableFuture<V> runAsync(String callableName, RetryStats retryStats, Callable<ListenableFuture<V>> callable)
     {
         requireNonNull(callableName, "callableName is null");
         requireNonNull(callable, "callable is null");
-        return runAsyncInternal(callableName, callable, new RetryStatus());
+        return runAsyncInternal(callableName, retryStats, callable, new RetryStatus());
     }
 
-    private <V> ListenableFuture<V> runAsyncInternal(String callableName, Callable<ListenableFuture<V>> callable, RetryStatus retryStatus)
+    private <V> ListenableFuture<V> runAsyncInternal(String callableName, RetryStats retryStats, Callable<ListenableFuture<V>> callable, RetryStatus retryStatus)
     {
         retryStatus.nextAttempt();
         ListenableFuture<V> resultFuture;
         try {
             resultFuture = callable.call();
+            retryStats.updateSuccess();
```

Comment:
I’d name all these “add” since “update” implies we are setting (overwriting) the value.